src/services/bundle_repository.js

```diff
@@ -78,14 +80,33 @@ export default class BundleRepository {
     return await downloadJSONFromGridFSBucket(bundleId, this.bundlesBucket);
   }
 
-  async createBundleMetadata(bundleId, storagePeriods, status = BundleStatusStates.unknown) {
+  async createBundleMetadata(bundleId, storagePeriods, status = BundleStatuses.unknown, additionalRepositoryFields = {}) {
     if (await this.db.collection('bundle_metadata').findOne({bundleId}) === null) {
       await this.db.collection('bundle_metadata').insertOne({
-        bundleId, storagePeriods, repository: {status}
+        bundleId, storagePeriods, repository: {...additionalRepositoryFields, status}
       });
     }
   }
 
+  async storeAdditionalMetadata(bundleId, complementaryMetadata) {
+    const requiredFields = [
```

Comment:
All required? Not too strict? What if only part of them will be downloaded? (I.E. from outdated node)

Reply:
Required was a bad name, changed